lib/babe/state.go

```diff
@@ -44,12 +45,16 @@ type BlockState interface {
 	GetFinalizedHeader(uint64, uint64) (*types.Header, error)
 	IsDescendantOf(parent, child common.Hash) (bool, error)
 	NumberIsFinalised(num *big.Int) (bool, error)
+	HandleRuntimeChanges(newState *rtstorage.TrieState, in runtime.Instance, bHash common.Hash) error
+	GetRuntime(*common.Hash) (runtime.Instance, error)
+	StoreRuntime(common.Hash, runtime.Instance)
 }
 
 // StorageState interface for storage state methods
 type StorageState interface {
 	TrieState(hash *common.Hash) (*rtstorage.TrieState, error)
 	StoreTrie(ts *rtstorage.TrieState, header *types.Header) error
+	LoadCodeHash(root *common.Hash) (common.Hash, error)
```

Comment:
same with this - can you also double check these in the `sync` package interfaces?

Reply:
Removed

Reply:
This is used in test. I have added an issue to remove this https://github.com/ChainSafe/gossamer/issues/1674

Reply:
if it's used in the test, I prefer casting the test state to the actual `*state.BlockState` and having the interface contain functions only used by the non-test code